dive_core/data_core.cpp

```diff
@@ -49,59 +49,84 @@ CaptureData::LoadResult DataCore::LoadCaptureData(const char *file_name)
-                                      true,
-                                      reserve_size,
-                                      m_log_ptr))
+    if (is_gfxr_capture)
     {
-        return false;
+        GfxrVulkanCommandHierarchyCreator vk_cmd_creator(m_capture_metadata.m_command_hierarchy,
+                                                         m_capture_data);
+        if (!vk_cmd_creator.CreateTrees())
+        {
+            return false;
+        }
+    }
+    else
+    {
+        std::unique_ptr<EmulateStateTracker> state_tracker(new EmulateStateTracker);
+
+        // Optional: Reserve the internal vectors based on the number of pm4 packets in the capture
+        // This is an educated guess that each PM4 packet results in x number of associated
+        // field/register nodes. Overguessing means more memory used during creation. Underguessing
+        // means more allocations. For big captures, this is easily in the multi-millions, so
+        // pre-reserving the space is a signficiant performance win
+        uint64_t reserve_size = m_capture_metadata.m_num_pm4_packets * 10;
+
+        // Command hierarchy tree creation
+        CommandHierarchyCreator cmd_hier_creator(*state_tracker);
+        if (!cmd_hier_creator.CreateTrees(&m_capture_metadata.m_command_hierarchy,
+                                          m_capture_data,
+                                          true,
+                                          reserve_size,
+                                          m_log_ptr))
+        {
+            return false;
+        }
     }
     return true;
 }
 
 //--------------------------------------------------------------------------------------------------
-bool DataCore::CreateMetaData()
+bool DataCore::CreateMetaData(bool is_gfxr_capture)
 {
-    std::unique_ptr<EmulateStateTracker> state_tracker(new EmulateStateTracker);
-    CaptureMetadataCreator               metadata_creator(m_capture_metadata, *state_tracker);
-    if (!metadata_creator.ProcessSubmits(m_capture_data.GetSubmits(),
-                                         m_capture_data.GetMemoryManager()))
+    if (!is_gfxr_capture)
     {
-        return false;
+        std::unique_ptr<EmulateStateTracker> state_tracker(new EmulateStateTracker);
+        CaptureMetadataCreator               metadata_creator(m_capture_metadata, *state_tracker);
+        if (!metadata_creator.ProcessSubmits(m_capture_data.GetSubmits(),
+                                             m_capture_data.GetMemoryManager()))
+        {
+            return false;
+        }
     }
     return true;
 }
 
 //--------------------------------------------------------------------------------------------------
-bool DataCore::ParseCaptureData()
+bool DataCore::ParseCaptureData(bool is_gfxr_capture)
 {
     if (m_progress_tracker)
     {
         m_progress_tracker->sendMessage("Processing command buffers...");
     }
 
-    if (!CreateMetaData())
+    if (is_gfxr_capture)
```

Comment:
i guess this can be simplified as 

```
        if (!CreateMetaData(is_gfxr_capture))
        {
            return false;
        }

        if (!CreateCommandHierarchy(is_gfxr_capture))
        {
            return false;
        }
```